cudaPackages: drop some runfile dependencies and fix a few builds

SomeoneSerge opened the pull request:
## Description of changes

A bunch of fixes aggregated from https://github.com/NixOS/nixpkgs/pull/271078 and https://github.com/NixOS/nixpkgs/pull/269639. Idk the other two PRs are going to be merged, but this specific diff should be mostly a no-brainer

@NixOS/cuda-maintainers

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

SomeoneSerge committed:
cudaPackagesGoogle: init, a package-set for jax and tf

SomeoneSerge committed:
tensorrt: dont break eval for unrelated packages

SomeoneSerge committed:
opensubdiv: drop the cudatoolkit.run file, and respect cudaFlags

SomeoneSerge committed:
blender: drop cudatoolkit.runfile

SomeoneSerge committed:
catboost: downgrade to cudaPackages_11 because of unsupported architectures (compute_35)

SomeoneSerge committed:
ctranslate2: fix the cuda 12 build

SomeoneSerge committed:
cudaPackages.cuda_nvcc: fix (getExe cuda_nvcc)

SomeoneSerge committed:
cudaPackages_12.cutensor: init and fix

SomeoneSerge committed:
python3Packages.cupy: fix (use older cutensor)

SomeoneSerge committed:
ucx: fix the cudaPackages_12 variant; drop the cudatoolkit runfile dependency

SomeoneSerge committed:
gromacs: drop cudatoolkit.run

SomeoneSerge committed:
openvino: opencvConfig.cmake attempts to find_package(CUDA)

SomeoneSerge committed:
nvidia-thrust: rm as deprecated

The GitHub repo has been archived, the new thing is
[cccl](https://github.com/nvidia/cccl)

SomeoneSerge committed:
ucc: drop the cudatoolkit runfile

SomeoneSerge committed:
ucc: respect cudaFlags

SomeoneSerge committed:
openmpi: drop the cudatoolkit runfile

SomeoneSerge committed:
python311Packages.torchWithCuda: drop cuda_cudart.static at runtime

SomeoneSerge committed:
cudaPackages.setupCudaHook: fix cudart flags

ConnorBaker commented:
I'm getting the following eval error from `Nixpkgs-review`:

```console
error:
       … while querying the derivation named 'dcgm-3.2.5'

       … while calling the 'getAttr' builtin

         at /derivation-internal.nix:19:19:

           18|       value = commonAttrs // {
           19|         outPath = builtins.getAttr outputName strict;
             |                   ^
           20|         drvPath = strict.drvPath;

       … while calling the 'derivationStrict' builtin

         at /derivation-internal.nix:9:12:

            8|
            9|   strict = derivationStrict drvAttrs;
             |            ^
           10|

       … while evaluating derivation 'dcgm-3.2.5'
         whose name attribute is located at /home/connorbaker/.cache/nixpkgs-review/pr-272063/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:348:7

       … while evaluating attribute 'EXTRA_CUDA_PATHS' of derivation 'dcgm-3.2.5'

         at /home/connorbaker/.cache/nixpkgs-review/pr-272063/nixpkgs/pkgs/os-specific/linux/dcgm/default.nix:98:3:

           97|   # Add our paths to the CUDA paths so FindCuda.cmake can find them.
           98|   EXTRA_CUDA_PATHS = lib.concatMapStringsSep "\n" mkAppendCudaPaths cudaPackageSetByVersion;
             |   ^
           99|   prePatch = ''

       … from call site

         at /home/connorbaker/.cache/nixpkgs-review/pr-272063/nixpkgs/pkgs/os-specific/linux/dcgm/default.nix:98:22:

           97|   # Add our paths to the CUDA paths so FindCuda.cmake can find them.
           98|   EXTRA_CUDA_PATHS = lib.concatMapStringsSep "\n" mkAppendCudaPaths cudaPackageSetByVersion;
             |                      ^
           99|   prePatch = ''

       … while calling 'concatMapStringsSep'

         at /home/connorbaker/.cache/nixpkgs-review/pr-272063/nixpkgs/lib/strings.nix:117:5:

          116|     # List of input strings
          117|     list: concatStringsSep sep (map f list);
             |     ^
          118|

       … while calling 'mkAppendCudaPaths'

         at /home/connorbaker/.cache/nixpkgs-review/pr-272063/nixpkgs/pkgs/os-specific/linux/dcgm/default.nix:65:23:

           64|   # Builds CMake code to add CUDA paths for include and lib.
           65|   mkAppendCudaPaths = { version, pkgSet }:
             |                       ^
           66|     let

       … while evaluating derivation 'cuda-combined-10'
         whose name attribute is located at /home/connorbaker/.cache/nixpkgs-review/pr-272063/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:348:7

       … while evaluating attribute 'paths' of derivation 'cuda-combined-10'

         at /home/connorbaker/.cache/nixpkgs-review/pr-272063/nixpkgs/pkgs/os-specific/linux/dcgm/default.nix:74:9:

           73|         name = "cuda-combined-${version}";
           74|         paths = pkgSet;
             |         ^
           75|       };

       … while evaluating derivation 'cudatoolkit-10.2.89'
         whose name attribute is located at /home/connorbaker/.cache/nixpkgs-review/pr-272063/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:348:7

       … while evaluating attribute 'depsTargetTargetPropagated' of derivation 'cudatoolkit-10.2.89'

         at /home/connorbaker/.cache/nixpkgs-review/pr-272063/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:403:7:

          402|       propagatedBuildInputs       = elemAt (elemAt propagatedDependencies 1) 1;
          403|       depsTargetTargetPropagated  = elemAt (elemAt propagatedDependencies 2) 0;
             |       ^
          404|

       … while evaluating derivation 'setup-cuda-hook'
         whose name attribute is located at /home/connorbaker/.cache/nixpkgs-review/pr-272063/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:348:7

       … while evaluating attribute 'cudartInclude' of derivation 'setup-cuda-hook'

         at /home/connorbaker/.cache/nixpkgs-review/pr-272063/nixpkgs/pkgs/development/compilers/cudatoolkit/extension.nix:72:11:

           71|           # - the static library, which is in the static output
           72|           substitutions.cudartInclude = "${final.cuda_cudart.dev}";
             |           ^
           73|           substitutions.cudartLib = "${final.cuda_cudart.lib}";

       error: attribute 'cuda_cudart' missing

       at /home/connorbaker/.cache/nixpkgs-review/pr-272063/nixpkgs/pkgs/development/compilers/cudatoolkit/extension.nix:72:44:

           71|           # - the static library, which is in the static output
           72|           substitutions.cudartInclude = "${final.cuda_cudart.dev}";
             |                                            ^
           73|           substitutions.cudartLib = "${final.cuda_cudart.lib}";
https://github.com/NixOS/nixpkgs/pull/272063 failed to build: Failed to list packages: nix-env failed with exit code 1
```

SomeoneSerge committed:
cudaPackages.setupCudaHook: rewrite cudartFlags, remove infinite recursion in cudatoolkit

We don't need to add the extra nvcc flags to locate cudart when using
cudatoolkit because it comes in the merged layout and nvcc doesn't have
any trouble locating dependencies in the same prefix

SomeoneSerge committed:
cudaPackages_11_3.saxpy: fallback to the cudatoolkit runfile

SomeoneSerge committed:
cudaPackages.cuda_nvcc: fix hook's offsets (-1, -1) -> (-1, 0)

Cf. explanations in https://github.com/NixOS/nixpkgs/pull/271078

SomeoneSerge committed:
cudaPackages.setupCudaHook: source only from nativeBuildInputs

SomeoneSerge commented:
From the commit message:


cudaPackages.setupCudaHook: rewrite cudartFlags, remove infinite recu…
…rsion in cudatoolkit

We don't need to add the extra nvcc flags to locate cudart when using
cudatoolkit because it comes in the merged layout and nvcc doesn't have
any trouble locating dependencies in the same prefix

SomeoneSerge commented:
- Colmap broken because it doesn't look at cudaFlags and the upstream hard-codes old architectures
- Gromacs seems to succeed locally

ConnorBaker commented:
Result of `nixpkgs-review pr 272063 --extra-nixpkgs-config '{
      allowUnfree = true;
      allowBroken = false;
      cudaSupport = true;
      cudaCapabilities = [ "8.9" ];
    }'` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>47 packages marked as broken and skipped:</summary>
  <ul>
    <li>caffeWithCuda</li>
    <li>caffeWithCuda.bin</li>
    <li>cudaPackages.nvidia_driver</li>
    <li>cudaPackages.nvidia_driver.bin</li>
    <li>cudaPackages.nvidia_driver.lib</li>
    <li>ezminc</li>
    <li>octavePackages.sparsersb</li>
    <li>python310Packages.caffeWithCuda</li>
    <li>python310Packages.caffeWithCuda.bin</li>
    <li>python310Packages.dm-sonnet</li>
    <li>python310Packages.dm-sonnet.dist</li>
    <li>python310Packages.elegy</li>
    <li>python310Packages.elegy.dist</li>
    <li>python310Packages.scikit-tda</li>
    <li>python310Packages.scikit-tda.dist</li>
    <li>python310Packages.sunpy</li>
    <li>python310Packages.sunpy.dist</li>
    <li>python310Packages.tensorflow-datasets</li>
    <li>python310Packages.tensorflow-datasets.dist</li>
    <li>python310Packages.theanoWithCuda</li>
    <li>python310Packages.theanoWithCuda.dist</li>
    <li>python310Packages.treex</li>
    <li>python310Packages.treex.dist</li>
    <li>python310Packages.worldengine</li>
    <li>python310Packages.worldengine.dist</li>
    <li>python311Packages.caffeWithCuda</li>
    <li>python311Packages.caffeWithCuda.bin</li>
    <li>python311Packages.dm-sonnet</li>
    <li>python311Packages.dm-sonnet.dist</li>
    <li>python311Packages.elegy</li>
    <li>python311Packages.elegy.dist</li>
    <li>python311Packages.scikit-tda</li>
    <li>python311Packages.scikit-tda.dist</li>
    <li>python311Packages.sunpy</li>
    <li>python311Packages.sunpy.dist</li>
    <li>python311Packages.tensorflow-datasets</li>
    <li>python311Packages.tensorflow-datasets.dist</li>
    <li>python311Packages.theanoWithCuda</li>
    <li>python311Packages.theanoWithCuda.dist</li>
    <li>python311Packages.treex</li>
    <li>python311Packages.treex.dist</li>
    <li>python311Packages.worldengine</li>
    <li>python311Packages.worldengine.dist</li>
    <li>t-rex</li>
    <li>truecrack-cuda</li>
    <li>worldengine-cli</li>
    <li>worldengine-cli.dist</li>
  </ul>
</details>
<details>
  <summary>46 packages failed to build:</summary>
  <ul>
    <li>colmapWithCuda</li>
    <li>cp2k</li>
    <li>cudaPackages.cuda-samples</li>
    <li>cudaPackages.nsight_compute (cudaPackages.nsight_compute.bin)</li>
    <li>dl-poly-classic-mpi</li>
    <li>getdp</li>
    <li>gromacsDouble</li>
    <li>gromacsDouble.dev</li>
    <li>gromacsDouble.man</li>
    <li>gromacsDoubleMpi</li>
    <li>gromacsDoubleMpi.dev</li>
    <li>gromacsDoubleMpi.man</li>
    <li>mathematica-cuda</li>
    <li>mathematica-webdoc-cuda</li>
    <li>petsc</li>
    <li>python310Packages.bsuite</li>
    <li>python310Packages.bsuite.dist</li>
    <li>python310Packages.gpt-2-simple</li>
    <li>python310Packages.gpt-2-simple.dist</li>
    <li>python310Packages.plotnine</li>
    <li>python310Packages.plotnine.dist</li>
    <li>python310Packages.python-mapnik</li>
    <li>python310Packages.python-mapnik.dist</li>
    <li>python310Packages.rlax</li>
    <li>python310Packages.rlax.dist</li>
    <li>python310Packages.tf2onnx</li>
    <li>python310Packages.tf2onnx.dist</li>
    <li>python310Packages.tiny-cuda-nn</li>
    <li>python311Packages.bsuite</li>
    <li>python311Packages.bsuite.dist</li>
    <li>python311Packages.gpt-2-simple</li>
    <li>python311Packages.gpt-2-simple.dist</li>
    <li>python311Packages.hoomd-blue</li>
    <li>python311Packages.plotnine</li>
    <li>python311Packages.plotnine.dist</li>
    <li>python311Packages.python-mapnik</li>
    <li>python311Packages.python-mapnik.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
    <li>python311Packages.tensorrt</li>
    <li>python311Packages.tensorrt.dist</li>
    <li>python311Packages.tf2onnx</li>
    <li>python311Packages.tf2onnx.dist</li>
    <li>python311Packages.tiny-cuda-nn</li>
    <li>saga</li>
    <li>tiny-cuda-nn</li>
  </ul>
</details>
<details>
  <summary>544 packages built:</summary>
  <ul>
    <li>EBTKS</li>
    <li>ants</li>
    <li>apacheHttpdPackages.mod_tile (apacheHttpdPackages_2_4.mod_tile)</li>
    <li>bicgl</li>
    <li>bicpl</li>
    <li>blender</li>
    <li>blender-hip</li>
    <li>c3d</li>
    <li>cdo</li>
    <li>cloudcompare</li>
    <li>conglomerate</li>
    <li>costa</li>
    <li>cudaPackages.backendStdenv</li>
    <li>cudaPackages.cuda_cccl (cudaPackages.cuda_cccl.dev)</li>
    <li>cudaPackages.cuda_cudart (cudaPackages.cuda_cudart.dev ,cudaPackages.cuda_cudart.lib ,cudaPackages.cuda_cudart.static)</li>
    <li>cudaPackages.cuda_cuobjdump (cudaPackages.cuda_cuobjdump.bin)</li>
    <li>cudaPackages.cuda_cupti (cudaPackages.cuda_cupti.dev ,cudaPackages.cuda_cupti.lib ,cudaPackages.cuda_cupti.sample ,cudaPackages.cuda_cupti.static)</li>
    <li>cudaPackages.cuda_cuxxfilt (cudaPackages.cuda_cuxxfilt.bin ,cudaPackages.cuda_cuxxfilt.dev ,cudaPackages.cuda_cuxxfilt.static)</li>
    <li>cudaPackages.cuda_demo_suite</li>
    <li>cudaPackages.cuda_documentation</li>
    <li>cudaPackages.cuda_gdb (cudaPackages.cuda_gdb.bin)</li>
    <li>cudaPackages.cuda_nsight (cudaPackages.cuda_nsight.bin)</li>
    <li>cudaPackages.cuda_nvcc (cudaPackages.cuda_nvcc.bin ,cudaPackages.cuda_nvcc.dev ,cudaPackages.cuda_nvcc.static)</li>
    <li>cudaPackages.cuda_nvdisasm (cudaPackages.cuda_nvdisasm.bin)</li>
    <li>cudaPackages.cuda_nvml_dev (cudaPackages.cuda_nvml_dev.dev ,cudaPackages.cuda_nvml_dev.lib)</li>
    <li>cudaPackages.cuda_nvprof (cudaPackages.cuda_nvprof.bin ,cudaPackages.cuda_nvprof.lib)</li>
    <li>cudaPackages.cuda_nvprune (cudaPackages.cuda_nvprune.bin)</li>
    <li>cudaPackages.cuda_nvrtc (cudaPackages.cuda_nvrtc.dev ,cudaPackages.cuda_nvrtc.lib ,cudaPackages.cuda_nvrtc.static)</li>
    <li>cudaPackages.cuda_nvtx (cudaPackages.cuda_nvtx.dev ,cudaPackages.cuda_nvtx.lib)</li>
    <li>cudaPackages.cuda_nvvp (cudaPackages.cuda_nvvp.bin)</li>
    <li>cudaPackages.cuda_opencl (cudaPackages.cuda_opencl.dev ,cudaPackages.cuda_opencl.lib)</li>
    <li>cudaPackages.cuda_profiler_api (cudaPackages.cuda_profiler_api.dev)</li>
    <li>cudaPackages.cuda_sanitizer_api (cudaPackages.cuda_sanitizer_api.bin)</li>
    <li>cudatoolkit (cudaPackages.cudatoolkit)</li>
    <li>cudatoolkit.doc (cudaPackages.cudatoolkit.doc)</li>
    <li>cudatoolkit.lib (cudaPackages.cudatoolkit.lib)</li>
    <li>cudaPackages.cudnn (cudaPackages.cudnn.dev ,cudaPackages.cudnn.lib ,cudaPackages.cudnn.static ,cudaPackages.cudnn_8_9 ,cudaPackages.cudnn_8_9.dev ,cudaPackages.cudnn_8_9.lib ,cudaPackages.cudnn_8_9.static)</li>
    <li>cudaPackages.cudnn_8_8 (cudaPackages.cudnn_8_8.dev ,cudaPackages.cudnn_8_8.lib ,cudaPackages.cudnn_8_8.static)</li>
    <li>cudaPackages.cutensor</li>
    <li>cudaPackages.cutensor.dev</li>
    <li>cudaPackages.fabricmanager (cudaPackages.fabricmanager.bin ,cudaPackages.fabricmanager.dev ,cudaPackages.fabricmanager.lib)</li>
    <li>cudaPackages.libcublas (cudaPackages.libcublas.dev ,cudaPackages.libcublas.lib ,cudaPackages.libcublas.static)</li>
    <li>cudaPackages.libcufft (cudaPackages.libcufft.dev ,cudaPackages.libcufft.lib ,cudaPackages.libcufft.static)</li>
    <li>cudaPackages.libcufile (cudaPackages.libcufile.dev ,cudaPackages.libcufile.lib ,cudaPackages.libcufile.sample ,cudaPackages.libcufile.static)</li>
    <li>cudaPackages.libcurand (cudaPackages.libcurand.dev ,cudaPackages.libcurand.lib ,cudaPackages.libcurand.static)</li>
    <li>cudaPackages.libcusolver (cudaPackages.libcusolver.dev ,cudaPackages.libcusolver.lib ,cudaPackages.libcusolver.static)</li>
    <li>cudaPackages.libcusparse (cudaPackages.libcusparse.dev ,cudaPackages.libcusparse.lib ,cudaPackages.libcusparse.static)</li>
    <li>cudaPackages.libnpp (cudaPackages.libnpp.dev ,cudaPackages.libnpp.lib ,cudaPackages.libnpp.static)</li>
    <li>cudaPackages.libnvidia_nscq (cudaPackages.libnvidia_nscq.bin ,cudaPackages.libnvidia_nscq.dev ,cudaPackages.libnvidia_nscq.lib)</li>
    <li>cudaPackages.libnvjitlink (cudaPackages.libnvjitlink.dev ,cudaPackages.libnvjitlink.lib ,cudaPackages.libnvjitlink.static)</li>
    <li>cudaPackages.libnvjpeg (cudaPackages.libnvjpeg.dev ,cudaPackages.libnvjpeg.lib ,cudaPackages.libnvjpeg.static)</li>
    <li>cudaPackages.libnvvm_samples</li>
    <li>cudaPackages.nccl</li>
    <li>cudaPackages.nccl-tests</li>
    <li>cudaPackages.nccl.dev</li>
    <li>cudaPackages.nsight_systems (cudaPackages.nsight_systems.bin)</li>
    <li>cudaPackages.nvidia_fs</li>
    <li>cudaPackages.saxpy</li>
    <li>cudaPackages.setupCudaHook</li>
    <li>cudaPackages.tensorrt (cudaPackages.tensorrt_8_6_1)</li>
    <li>cudaPackages.tensorrt.dev (cudaPackages.tensorrt_8_6_1.dev)</li>
    <li>cudatoolkit_11</li>
    <li>cudatoolkit_11.doc</li>
    <li>cudatoolkit_11.lib</li>
    <li>dbcsr</li>
    <li>dcgm</li>
    <li>dtcmp</li>
    <li>dtcmp.dev</li>
    <li>eccodes</li>
    <li>elastix</li>
    <li>elmerfem</li>
    <li>elpa</li>
    <li>elpa.dev</li>
    <li>elpa.doc</li>
    <li>elpa.man</li>
    <li>entwine</li>
    <li>faissWithCuda</li>
    <li>faissWithCuda.demos</li>
    <li>febio</li>
    <li>fftwMpi</li>
    <li>fftwMpi.dev</li>
    <li>fftwMpi.info</li>
    <li>fftwMpi.man</li>
    <li>freecad</li>
    <li>frigate</li>
    <li>gdal (python311Packages.gdal)</li>
    <li>globalarrays</li>
    <li>gmt</li>
    <li>gnudatalanguage</li>
    <li>gplates</li>
    <li>gpt2tc</li>
    <li>gpu-burn</li>
    <li>grass</li>
    <li>gromacs</li>
    <li>gromacs.dev</li>
    <li>gromacs.man</li>
    <li>gromacsMpi (gromacsCudaMpi)</li>
    <li>gromacsMpi.dev (gromacsCudaMpi.dev)</li>
    <li>gromacsMpi.man (gromacsCudaMpi.man)</li>
    <li>gromacsPlumed</li>
    <li>gromacsPlumed.dev</li>
    <li>gromacsPlumed.man</li>
    <li>gwe</li>
    <li>haskellPackages.haskell-mpi</li>
    <li>haskellPackages.haskell-mpi.doc</li>
    <li>haskellPackages.hgdal</li>
    <li>haskellPackages.hgdal.doc</li>
    <li>haskellPackages.mpi-hs-binary</li>
    <li>haskellPackages.mpi-hs-binary.doc</li>
    <li>haskellPackages.mpi-hs-cereal</li>
    <li>haskellPackages.mpi-hs-cereal.doc</li>
    <li>haskellPackages.mpi-hs-store</li>
    <li>haskellPackages.mpi-hs-store.doc</li>
    <li>hdf5-mpi</li>
    <li>hdf5-mpi.dev</li>
    <li>highfive-mpi</li>
    <li>hp2p</li>
    <li>hpcg</li>
    <li>hpl</li>
    <li>inormalize</li>
    <li>intensity-normalization (python311Packages.intensity-normalization)</li>
    <li>intensity-normalization.dist (python311Packages.intensity-normalization.dist)</li>
    <li>ior</li>
    <li>itk (itk_5)</li>
    <li>itk_5_2</li>
    <li>katagoTensorRT</li>
    <li>katagoWithCuda</li>
    <li>labplot</li>
    <li>lammps-mpi</li>
    <li>libcircle</li>
    <li>libcircle.dev</li>
    <li>libminc</li>
    <li>librealsenseWithCuda</li>
    <li>librealsenseWithCuda.dev</li>
    <li>librsb</li>
    <li>libtensorflow (libtensorflow.python)</li>
    <li>libvdwxc</li>
    <li>lwgrp</li>
    <li>lwgrp.dev</li>
    <li>magma-cuda</li>
    <li>magma-cuda-static</li>
    <li>mapcache</li>
    <li>mapnik</li>
    <li>mapproxy</li>
    <li>mapproxy.dist</li>
    <li>mapserver</li>
    <li>merkaartor</li>
    <li>migrate</li>
    <li>minc_tools</li>
    <li>minc_widgets</li>
    <li>mirtk</li>
    <li>mni_autoreg</li>
    <li>mpi (openmpi)</li>
    <li>mpi.man (openmpi.man)</li>
    <li>mpifileutils</li>
    <li>mpifileutils.dev</li>
    <li>mpifileutils.man</li>
    <li>mrtrix</li>
    <li>mysql-workbench</li>
    <li>n3</li>
    <li>nco</li>
    <li>ncview</li>
    <li>nest-mpi</li>
    <li>netcdf</li>
    <li>netcdf-mpi</li>
    <li>netcdfcxx4</li>
    <li>netcdffortran</li>
    <li>neuron-full (python311Packages.neuronpy)</li>
    <li>neuron-mpi</li>
    <li>nvtop</li>
    <li>nvtop-nvidia</li>
    <li>nwchem</li>
    <li>octavePackages.mapping</li>
    <li>octavePackages.ncarray</li>
    <li>octavePackages.netcdf</li>
    <li>octopus</li>
    <li>oobicpl</li>
    <li>openorienteering-mapper</li>
    <li>opensubdiv</li>
    <li>opensubdiv.dev</li>
    <li>ovito</li>
    <li>paraview</li>
    <li>parmetis</li>
    <li>pdal</li>
    <li>perl536Packages.Tirex</li>
    <li>perl536Packages.Tirex.devdoc</li>
    <li>perl538Packages.Tirex</li>
    <li>perl538Packages.Tirex.devdoc</li>
    <li>postgresql12JitPackages.postgis</li>
    <li>postgresql12JitPackages.postgis.doc</li>
    <li>postgresql12Packages.postgis</li>
    <li>postgresql12Packages.postgis.doc</li>
    <li>postgresql13JitPackages.postgis</li>
    <li>postgresql13JitPackages.postgis.doc</li>
    <li>postgresql13Packages.postgis</li>
    <li>postgresql13Packages.postgis.doc</li>
    <li>postgresql14JitPackages.postgis</li>
    <li>postgresql14JitPackages.postgis.doc</li>
    <li>postgresql14Packages.postgis</li>
    <li>postgresql14Packages.postgis.doc</li>
    <li>postgresqlJitPackages.postgis (postgresql15JitPackages.postgis)</li>
    <li>postgresqlJitPackages.postgis.doc (postgresql15JitPackages.postgis.doc)</li>
    <li>postgresqlPackages.postgis (postgresql15Packages.postgis)</li>
    <li>postgresqlPackages.postgis.doc (postgresql15Packages.postgis.doc)</li>
    <li>postgresql16JitPackages.postgis</li>
    <li>postgresql16JitPackages.postgis.doc</li>
    <li>postgresql16Packages.postgis</li>
    <li>postgresql16Packages.postgis.doc</li>
    <li>precice</li>
    <li>prometheus-dcgm-exporter</li>
    <li>pymol</li>
    <li>pymol.dist</li>
    <li>python310Packages.arviz</li>
    <li>python310Packages.arviz.dist</li>
    <li>python310Packages.bambi</li>
    <li>python310Packages.bambi.dist</li>
    <li>python310Packages.baselines</li>
    <li>python310Packages.baselines.dist</li>
    <li>python310Packages.bitsandbytes</li>
    <li>python310Packages.bitsandbytes.dist</li>
    <li>python310Packages.boltztrap2</li>
    <li>python310Packages.boltztrap2.dist</li>
    <li>python310Packages.cartopy</li>
    <li>python310Packages.cartopy.dist</li>
    <li>python310Packages.cleanlab</li>
    <li>python310Packages.cleanlab.dist</li>
    <li>python310Packages.cupy</li>
    <li>python310Packages.cupy.dist</li>
    <li>python310Packages.dalle-mini</li>
    <li>python310Packages.dalle-mini.dist</li>
    <li>python310Packages.dask-mpi</li>
    <li>python310Packages.dask-mpi.dist</li>
    <li>python310Packages.datashader</li>
    <li>python310Packages.datashader.dist</li>
    <li>python310Packages.distrax</li>
    <li>python310Packages.distrax.dist</li>
    <li>python310Packages.django-bootstrap4</li>
    <li>python310Packages.django-bootstrap4.dist</li>
    <li>python310Packages.eccodes</li>
    <li>python310Packages.edward</li>
    <li>python310Packages.edward.dist</li>
    <li>python310Packages.encodec</li>
    <li>python310Packages.encodec.dist</li>
    <li>python310Packages.fenics</li>
    <li>python310Packages.fenics.dist</li>
    <li>python310Packages.fiona</li>
    <li>python310Packages.fiona.dist</li>
    <li>python310Packages.fipy</li>
    <li>python310Packages.fipy.dist</li>
    <li>python310Packages.flax</li>
    <li>python310Packages.flax.dist</li>
    <li>python310Packages.folium</li>
    <li>python310Packages.folium.dist</li>
    <li>python310Packages.gdal</li>
    <li>python310Packages.geopandas</li>
    <li>python310Packages.geopandas.dist</li>
    <li>python310Packages.gpaw</li>
    <li>python310Packages.gpaw.dist</li>
    <li>python310Packages.h5netcdf</li>
    <li>python310Packages.h5netcdf.dist</li>
    <li>python310Packages.h5py-mpi</li>
    <li>python310Packages.h5py-mpi.dist</li>
    <li>python310Packages.hoomd-blue</li>
    <li>python310Packages.intensity-normalization</li>
    <li>python310Packages.intensity-normalization.dist</li>
    <li>python310Packages.jaxlibWithCuda</li>
    <li>python310Packages.jaxlibWithCuda.dist</li>
    <li>python310Packages.lightgbm</li>
    <li>python310Packages.lightgbm.dist</li>
    <li>python310Packages.meep</li>
    <li>python310Packages.meshio</li>
    <li>python310Packages.meshio.dist</li>
    <li>python310Packages.mhcflurry</li>
    <li>python310Packages.mhcflurry.dist</li>
    <li>python310Packages.mpi4py</li>
    <li>python310Packages.mpi4py.dist</li>
    <li>python310Packages.n3fit</li>
    <li>python310Packages.n3fit.dist</li>
    <li>python310Packages.nest</li>
    <li>python310Packages.netcdf4</li>
    <li>python310Packages.netcdf4.dist</li>
    <li>python310Packages.neuronpy</li>
    <li>python310Packages.numbaWithCuda</li>
    <li>python310Packages.numbaWithCuda.dist</li>
    <li>python310Packages.objax</li>
    <li>python310Packages.objax.dist</li>
    <li>python310Packages.openai-triton-bin</li>
    <li>python310Packages.openai-triton-bin.dist</li>
    <li>python310Packages.openai-triton-cuda</li>
    <li>python310Packages.openai-triton-cuda.dist</li>
    <li>python310Packages.openllm</li>
    <li>python310Packages.openllm.dist</li>
    <li>python310Packages.openusd</li>
    <li>python310Packages.openusd.doc</li>
    <li>python310Packages.osmnx</li>
    <li>python310Packages.osmnx.dist</li>
    <li>python310Packages.pot</li>
    <li>python310Packages.pot.dist</li>
    <li>python310Packages.pycuda</li>
    <li>python310Packages.pycuda.dist</li>
    <li>python310Packages.pydicom-seg</li>
    <li>python310Packages.pydicom-seg.dist</li>
    <li>python310Packages.pygmt</li>
    <li>python310Packages.pygmt.dist</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.pymc.dist</li>
    <li>python310Packages.pymedio</li>
    <li>python310Packages.pymedio.dist</li>
    <li>python310Packages.pynvml</li>
    <li>python310Packages.pynvml.dist</li>
    <li>python310Packages.pyprecice</li>
    <li>python310Packages.pyprecice.dist</li>
    <li>python310Packages.pyradiomics</li>
    <li>python310Packages.pyradiomics.dist</li>
    <li>python310Packages.pyrealsense2WithCuda</li>
    <li>python310Packages.pyrealsense2WithCuda.dev</li>
    <li>python310Packages.pytensor</li>
    <li>python310Packages.pytensor.dist</li>
    <li>python310Packages.qcodes</li>
    <li>python310Packages.qcodes-contrib-drivers</li>
    <li>python310Packages.qcodes-contrib-drivers.dist</li>
    <li>python310Packages.qcodes.dist</li>
    <li>python310Packages.rasterio</li>
    <li>python310Packages.rasterio.dist</li>
    <li>python310Packages.sfepy</li>
    <li>python310Packages.sfepy.dist</li>
    <li>python310Packages.shap</li>
    <li>python310Packages.shap.dist</li>
    <li>python310Packages.simpleitk</li>
    <li>python310Packages.simpleitk.dist</li>
    <li>python310Packages.tensorflow (python310Packages.tensorflow-build ,python310Packages.tensorflowWithCuda)</li>
    <li>python310Packages.tensorflow.dist (python310Packages.tensorflow-build.dist ,python310Packages.tensorflowWithCuda.dist)</li>
    <li>python310Packages.tensorflow-probability</li>
    <li>python310Packages.tensorflow-probability.dist</li>
    <li>python310Packages.tensorflowWithoutCuda</li>
    <li>python310Packages.tensorflowWithoutCuda.dist</li>
    <li>python310Packages.tensorrt</li>
    <li>python310Packages.tensorrt.dist</li>
    <li>python310Packages.tflearn</li>
    <li>python310Packages.tflearn.dist</li>
    <li>python310Packages.torchWithCuda</li>
    <li>python310Packages.torchWithCuda.dev</li>
    <li>python310Packages.torchWithCuda.dist</li>
    <li>python310Packages.torchWithCuda.lib</li>
    <li>python310Packages.torchaudio-bin</li>
    <li>python310Packages.torchaudio-bin.dist</li>
    <li>python310Packages.torchio</li>
    <li>python310Packages.torchio.dist</li>
    <li>python310Packages.torchvision-bin</li>
    <li>python310Packages.torchvision-bin.dist</li>
    <li>python310Packages.trfl</li>
    <li>python310Packages.trfl.dist</li>
    <li>python310Packages.umap-learn</li>
    <li>python310Packages.umap-learn.dist</li>
    <li>python310Packages.vqgan-jax</li>
    <li>python310Packages.vqgan-jax.dist</li>
    <li>python310Packages.wktutils</li>
    <li>python310Packages.wktutils.dist</li>
    <li>python310Packages.wrf-python</li>
    <li>python310Packages.wrf-python.dist</li>
    <li>python311Packages.arviz</li>
    <li>python311Packages.arviz.dist</li>
    <li>python311Packages.bambi</li>
    <li>python311Packages.bambi.dist</li>
    <li>python311Packages.baselines</li>
    <li>python311Packages.baselines.dist</li>
    <li>python311Packages.bitsandbytes</li>
    <li>python311Packages.bitsandbytes.dist</li>
    <li>python311Packages.boltztrap2</li>
    <li>python311Packages.boltztrap2.dist</li>
    <li>python311Packages.cartopy</li>
    <li>python311Packages.cartopy.dist</li>
    <li>python311Packages.cleanlab</li>
    <li>python311Packages.cleanlab.dist</li>
    <li>python311Packages.cupy</li>
    <li>python311Packages.cupy.dist</li>
    <li>python311Packages.dalle-mini</li>
    <li>python311Packages.dalle-mini.dist</li>
    <li>python311Packages.dask-mpi</li>
    <li>python311Packages.dask-mpi.dist</li>
    <li>python311Packages.datashader</li>
    <li>python311Packages.datashader.dist</li>
    <li>python311Packages.distrax</li>
    <li>python311Packages.distrax.dist</li>
    <li>python311Packages.django-bootstrap4</li>
    <li>python311Packages.django-bootstrap4.dist</li>
    <li>python311Packages.eccodes</li>
    <li>python311Packages.edward</li>
    <li>python311Packages.edward.dist</li>
    <li>python311Packages.encodec</li>
    <li>python311Packages.encodec.dist</li>
    <li>python311Packages.fenics</li>
    <li>python311Packages.fenics.dist</li>
    <li>python311Packages.fiona</li>
    <li>python311Packages.fiona.dist</li>
    <li>python311Packages.fipy</li>
    <li>python311Packages.fipy.dist</li>
    <li>python311Packages.flax</li>
    <li>python311Packages.flax.dist</li>
    <li>python311Packages.folium</li>
    <li>python311Packages.folium.dist</li>
    <li>python311Packages.geopandas</li>
    <li>python311Packages.geopandas.dist</li>
    <li>python311Packages.gpaw</li>
    <li>python311Packages.gpaw.dist</li>
    <li>python311Packages.h5netcdf</li>
    <li>python311Packages.h5netcdf.dist</li>
    <li>python311Packages.h5py-mpi</li>
    <li>python311Packages.h5py-mpi.dist</li>
    <li>python311Packages.jaxlibWithCuda</li>
    <li>python311Packages.jaxlibWithCuda.dist</li>
    <li>python311Packages.lightgbm</li>
    <li>python311Packages.lightgbm.dist</li>
    <li>python311Packages.meep</li>
    <li>python311Packages.meshio</li>
    <li>python311Packages.meshio.dist</li>
    <li>python311Packages.mhcflurry</li>
    <li>python311Packages.mhcflurry.dist</li>
    <li>python311Packages.mpi4py</li>
    <li>python311Packages.mpi4py.dist</li>
    <li>python311Packages.n3fit</li>
    <li>python311Packages.n3fit.dist</li>
    <li>python311Packages.nest</li>
    <li>python311Packages.netcdf4</li>
    <li>python311Packages.netcdf4.dist</li>
    <li>python311Packages.numbaWithCuda</li>
    <li>python311Packages.numbaWithCuda.dist</li>
    <li>python311Packages.objax</li>
    <li>python311Packages.objax.dist</li>
    <li>python311Packages.openai-triton-bin</li>
    <li>python311Packages.openai-triton-bin.dist</li>
    <li>python311Packages.openai-triton-cuda</li>
    <li>python311Packages.openai-triton-cuda.dist</li>
    <li>python311Packages.openllm</li>
    <li>python311Packages.openllm.dist</li>
    <li>python311Packages.openusd</li>
    <li>python311Packages.openusd.doc</li>
    <li>python311Packages.osmnx</li>
    <li>python311Packages.osmnx.dist</li>
    <li>python311Packages.pot</li>
    <li>python311Packages.pot.dist</li>
    <li>python311Packages.pycuda</li>
    <li>python311Packages.pycuda.dist</li>
    <li>python311Packages.pydicom-seg</li>
    <li>python311Packages.pydicom-seg.dist</li>
    <li>python311Packages.pygmt</li>
    <li>python311Packages.pygmt.dist</li>
    <li>python311Packages.pymc</li>
    <li>python311Packages.pymc.dist</li>
    <li>python311Packages.pymedio</li>
    <li>python311Packages.pymedio.dist</li>
    <li>python311Packages.pynvml</li>
    <li>python311Packages.pynvml.dist</li>
    <li>python311Packages.pyprecice</li>
    <li>python311Packages.pyprecice.dist</li>
    <li>python311Packages.pyradiomics</li>
    <li>python311Packages.pyradiomics.dist</li>
    <li>python311Packages.pyrealsense2WithCuda</li>
    <li>python311Packages.pyrealsense2WithCuda.dev</li>
    <li>python311Packages.pytensor</li>
    <li>python311Packages.pytensor.dist</li>
    <li>python311Packages.qcodes</li>
    <li>python311Packages.qcodes-contrib-drivers</li>
    <li>python311Packages.qcodes-contrib-drivers.dist</li>
    <li>python311Packages.qcodes.dist</li>
    <li>python311Packages.rasterio</li>
    <li>python311Packages.rasterio.dist</li>
    <li>python311Packages.sfepy</li>
    <li>python311Packages.sfepy.dist</li>
    <li>python311Packages.shap</li>
    <li>python311Packages.shap.dist</li>
    <li>python311Packages.simpleitk</li>
    <li>python311Packages.simpleitk.dist</li>
    <li>python311Packages.tensorflow (python311Packages.tensorflow-build ,python311Packages.tensorflowWithCuda)</li>
    <li>python311Packages.tensorflow.dist (python311Packages.tensorflow-build.dist ,python311Packages.tensorflowWithCuda.dist)</li>
    <li>python311Packages.tensorflow-probability</li>
    <li>python311Packages.tensorflow-probability.dist</li>
    <li>python311Packages.tensorflowWithoutCuda</li>
    <li>python311Packages.tensorflowWithoutCuda.dist</li>
    <li>python311Packages.tflearn</li>
    <li>python311Packages.tflearn.dist</li>
    <li>python311Packages.torchWithCuda</li>
    <li>python311Packages.torchWithCuda.dev</li>
    <li>python311Packages.torchWithCuda.dist</li>
    <li>python311Packages.torchWithCuda.lib</li>
    <li>python311Packages.torchaudio-bin</li>
    <li>python311Packages.torchaudio-bin.dist</li>
    <li>python311Packages.torchio</li>
    <li>python311Packages.torchio.dist</li>
    <li>python311Packages.torchvision-bin</li>
    <li>python311Packages.torchvision-bin.dist</li>
    <li>python311Packages.trfl</li>
    <li>python311Packages.trfl.dist</li>
    <li>python311Packages.umap-learn</li>
    <li>python311Packages.umap-learn.dist</li>
    <li>python311Packages.vqgan-jax</li>
    <li>python311Packages.vqgan-jax.dist</li>
    <li>python311Packages.wktutils</li>
    <li>python311Packages.wktutils.dist</li>
    <li>python311Packages.wrf-python</li>
    <li>python311Packages.wrf-python.dist</li>
    <li>pytrainer</li>
    <li>pytrainer.dist</li>
    <li>qgis</li>
    <li>qgis-ltr</li>
    <li>qmapshack</li>
    <li>quantum-espresso-mpi</li>
    <li>raxml-mpi</li>
    <li>rocmPackages.rocalution (rocmPackages_5.rocalution)</li>
    <li>rocmPackages.rocprofiler (rocmPackages_5.rocprofiler)</li>
    <li>scalapack</li>
    <li>scalapack.dev</li>
    <li>scotch</li>
    <li>siesta-mpi</li>
    <li>simpleitk</li>
    <li>sirius</li>
    <li>spfft</li>
    <li>spla</li>
    <li>sumo</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ompi</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ompi-c</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ompi-cxx</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ompi-f77</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ompi-f90</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ompi-fort</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.orte</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.tensorflow</li>
    <li>trilinos-mpi</li>
    <li>tts</li>
    <li>tts.dist</li>
    <li>ucc</li>
    <li>ucc.dev</li>
    <li>ucx</li>
    <li>ucx.dev</li>
    <li>ucx.doc</li>
    <li>udig</li>
    <li>vpv</li>
    <li>wyoming-openwakeword</li>
    <li>wyoming-openwakeword.dist</li>
    <li>xgboostWithCuda</li>
    <li>xpraWithNvenc</li>
    <li>xpraWithNvenc.dist</li>
    <li>xyce-parallel</li>
    <li>xyce-parallel.doc</li>
  </ul>
</details>

ConnorBaker commented on pkgs/development/compilers/cudatoolkit/extension.nix:
```diff
@@ -54,11 +54,28 @@ final: prev: let
+            builtins.concatStringsSep " " (final.lib.optionals (final ? cuda_cudart) ([
+              "-I${final.lib.getDev cudart}/include"
+              "-L${final.lib.getLib cudart}/lib"
+            ] ++ final.lib.optionals (builtins.elem "lib" cudart.outputs) [
```
```suggestion
            ] ++ final.lib.optionals (cudart ? static) [
```

you use the `static` output, so it seems odd you check for the `lib` output. (Recall that with the individually-packaged CUDA redistributables `lib` contains dynamic libraries and `static` contains static libraries.)

SomeoneSerge replied:
Typo, meant to check for 'static" and use `cudart.static`

ConnorBaker commented on pkgs/development/compilers/cudatoolkit/hooks/setup-cuda-hook.sh:
```diff
@@ -55,8 +58,9 @@ setupCUDAToolkitCompilers() {
-    if [[ -z "${nvccDontPrependCudartFlags-}" ]] ; then
-        export NVCC_APPEND_FLAGS+=" -L@cudartLib@/lib -L@cudartStatic@/lib -I@cudartInclude@/include"
+    local cudartFlags="@cudartFlags@"
+    if [[ -z "${nvccDontPrependCudartFlags-}" ]] && [[ -n "\${cudartFlags:-}" ]] ; then
```
How is `"\${cudartFlags:-}"` interpreted (if it is) given `$` is escaped? I would think that this test always evaluates to false because there's no string interpolation happening and it's not empty.

SomeoneSerge replied:
Holy mother, I must've copied that from a nix expression

SomeoneSerge committed:
tiny-cuda-nn: cuda_cccl required with the newer cuda

SomeoneSerge committed:
tiny-cuda-nn: prune runtime closure

ConnorBaker commented on pkgs/top-level/all-packages.nix:
```diff
@@ -7318,9 +7318,13 @@ with pkgs;
+  # compatibility: https://www.tensorflow.org/install/source#gpu
+  cudaPackagesGoogle = cudaPackages_11;
+
   # TODO: try upgrading once there is a cuDNN release supporting CUDA 12. No
   # such cuDNN release as of 2023-01-10.
```
Remove this as we're doing the bump here

SomeoneSerge replied:
The bump is in the other PR?

SomeoneSerge commented:
https://github.com/NixOS/nixpkgs/pull/272964 cc @janydoe @ConnorBaker
